kong/router/init.lua

```diff
@@ -30,7 +30,137 @@ function _M:select(req_method, req_uri, req_host, req_scheme,
 end
 
 
+
+local phonehome_statistics
+do
+  local reports = require("kong.reports")
+  local nkeys = require("table.nkeys")
+  local log = ngx.log
+  local ERR = ngx.ERR
+
+  local byte = string.byte
+  local TILDE = byte("~")
+  local function is_regex_magic(path)
+    return byte(path) == TILDE
+  end
+
+  local empty_table = {}
+  -- reuse tables to avoid cost of creating tables and garbage collection
+  local protocols = {
+    http = 0, -- { "http", "https" },
+    stream = 0, -- { "tcp", "tls", "udp" },
+    tls_passthrough = 0, -- { "tls_passthrough" },
+    grpc = 0, -- { "grpc", "grpcs" },
+  }
+  local path_handlings = {
+    v0 = 0,
+    v1 = 0,
+  }
+  local route_report = {
+    flavor = "unknown",
+    paths = 0,
+    headers = 0,
+    routes = 0,
+    regex_routes = 0,
+    protocols = protocols,
+    path_handlings = path_handlings,
+  }
+
+  local function traditional_statistics(routes)
+    local paths_count = 0
+    local headers_count = 0
+    local regex_paths_count = 0
+    local http = 0
+    local stream = 0
+    local tls_passthrough = 0
+    local grpc = 0
+    local v0 = 0
+    local v1 = 0
+
+    for _, route in ipairs(routes) do
```

Comment:
rename this `route` to `r`, then the next line will be `route = r.route`, it is more readable.